tests/integration/ha_tests/helpers.py

```diff
@@ -642,15 +642,13 @@ async def get_primary(ops_test: OpsTest, app, down_unit: str = None) -> str:
     """
     for unit in ops_test.model.applications[app].units:
         if unit.name != down_unit:
-            break
```

Comment:
change this helper function to match the behavior on K8s part, running the action in each unit once, instead of retrying multiple times on the same unit.